src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -50,12 +75,28 @@ String profileOwnerName = (String) request.getAttribute("profileOwner");
         <button type="submit">Submit</button>
       </form>
     <% } %>
-  </div>
+    <h2><%= profileOwnerName %>'s sent messages</h2>
 
-  <div id="container">
-    <h1>Messages will go here?</h1>
+    <!--This is where the user's sent messages will show up -->
+    <div id="messages">
+      <ul>
+    <% //This queue acts as a priority queue. Everytime you pop, the oldest message is returned
+       Queue<Message> user_messages = MessageStore.getInstance()
+                                         .getMessagesByOwner(profileOwnerId); 
+    %>
+    <% int initial_size = user_messages.size(); %>
+    <% for (int i = 0; i < initial_size; ++i) { %>
+    <%   // If the message is not empty, then print it %>
+    <%   if(user_messages.peek().getContent() != null && 
+                                              !user_messages.peek().getContent().isEmpty()) { %>
+          <li> <b> <%= formatter.format(Date.from(user_messages.peek().getCreationTime())) %> 
+          </b>: <%= user_messages.peek().getContent() %> </li>
+    <%   } %>      
+    <%   user_messages.remove(); %>
```

Comment:
After reading through this, I'm not sure it's necessary to have a PQ here as opposed to a list that has been sorted. Because you are only setting the data once and never update it after that, I'd lean towards a sorted list. Priority Queues tend to be better when updates need to be made as they will maintain an ordering over time.